src/charm.py

```diff
@@ -2,77 +2,71 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+"""Charmed Machine Operator for Apache ZooKeeper."""
+
 import logging
-import subprocess
 
+from charms.kafka.v0.kafka_snap import KafkaSnap
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
-
-from kafka_helpers import install_packages, merge_config
+from ops.model import Relation
 
 logger = logging.getLogger(__name__)
 
 
-class KafkaCharm(CharmBase):
+class ZooKeeperCharm(CharmBase):
+    """Charmed Operator for ZooKeeper."""
+
     def __init__(self, *args):
         super().__init__(*args)
-        self.name = "kafka"
+        self.name = "zookeeper"
+        self.snap = KafkaSnap()
 
         self.framework.observe(getattr(self.on, "install"), self._on_install)
         self.framework.observe(
             getattr(self.on, "cluster_relation_joined"), self._on_cluster_relation_joined
         )
         self.framework.observe(getattr(self.on, "leader_elected"), self._on_leader_elected)
         self.framework.observe(
-            getattr(self.on, "get_server_properties_action"), self._on_get_server_properties_action
+            getattr(self.on, "get_zookeeper_properties_action"), self._on_get_properties_action
         )
+        self.framework.observe(
+            getattr(self.on, "get_snap_apps_action"), self._on_get_snap_apps_action
+        )
+        self.framework.observe(getattr(self.on, "config_changed"), self._on_config_changed)
 
     @property
-    def _relation(self):
+    def _relation(self) -> Relation:
         return self.model.get_relation("cluster")
 
     def _on_install(self, _) -> None:
-        try:
-            self.unit.status = MaintenanceStatus("installing packages")
-            install_packages()
-            self.unit.status = ActiveStatus()
-        except:
-            self.unit.status = BlockedStatus("failed to install packages")
-
-    def _on_leader_elected(self, _):
-        return
+        """Handler for on_install event."""
+        self.unit.status = self.snap.install_kafka_snap()
```

Comment:
I know the happy path of this is setting MaintenanceStatus, but isn't more sensible to set the MaintenanceStatus at the start of the install process, instead of being the result of it?

Reply:
The default `install` hook has `MaintenanceStatus: (install) installing charm software`, which tbh covers my bases!